lib/services/calculation_service.dart

```diff
@@ -0,0 +1,23 @@
+import 'interfaces/calculation_service_interface.dart';
+
+class CalculationService extends ICalculationService {
+  int calculateAge({DateTime birthDate, String birthDateString}) {
+    if (birthDateString != null && birthDate == null) {
+      birthDate = DateTime.parse(birthDateString);
+    }
+    var currentDate = DateTime.now();
+    var age = currentDate.year - birthDate.year;
+    var month1 = currentDate.month;
+    var month2 = birthDate.month;
+    if (month2 > month1) {
+      age--;
+    } else if (month1 == month2) {
+      var day1 = currentDate.day;
+      var day2 = birthDate.day;
+      if (day2 > day1) {
+        age--;
+      }
+    }
+    return age;
```

Comment:
too complex logic, would be easier with this
```ts
function getAge(d1, d2){
    d2 = d2 || new Date();
    var diff = d2.getTime() - d1.getTime();
    return Math.floor(diff / (1000 * 60 * 60 * 24 * 365.25));
}
```
refs: https://stackoverflow.com/questions/4076321/javascript-age-calculation

Reply:
NOPE. Dart cannot compare 2 DateTime and get its result to calculate in mathematic way.